openedx/code-annotations
# Simple local report rendering

**Description:** 

Finishes the ancient implementation of human readable reporting in rst and html. This should allow people to get a readable report from any well-configured annotation output without having to deal with Sphinx.

Closes: #128 

**Testing instructions:**

1. Create a code-annotations in any of the usual ways or use the one here: https://gist.github.com/bmtcril/0441b96e8cc92ba0c24b3178c4ef8e00
2. See if the instructions in the getting_started and configuration docs are sufficient to help you configure and run the `generate_docs` command. A config that works with the gist above is included in that gist.
3. Try viewing rst documents using `restview` (linked in the docs)
4. Try viewing html document using a browser

Docs generate cleanly, should not throw errors when viewed, and should be relatively easy to follow. Links to github should work if configured.

restview screenshots:

![Screenshot 2024-12-19 at 4 21 04 PM](https://github.com/user-attachments/assets/8bb115cf-4a1f-4760-b738-5bb9d7089bc3)
![Screenshot 2024-12-19 at 4 21 45 PM](https://github.com/user-attachments/assets/e0eaec45-63d2-46a5-a514-91eae60e0755)

HTML screenshots:

![Screenshot 2024-12-19 at 4 22 09 PM](https://github.com/user-attachments/assets/f5472c76-845a-42eb-a1fb-f896f2f98841)
![Screenshot 2024-12-19 at 4 22 39 PM](https://github.com/user-attachments/assets/afd573bd-c9e5-43d0-a1e0-2ed69fdecb94)
